### PR TITLE
Port PR merge script from Spark

### PR DESCRIPTION
**Pull request by JoshRosen**
This ports the PR merge script from Spark, similar to what https://github.com/databricks/spark-avro/pull/81 did for spark-avro.

@falaki, let's use this script from now on when merging PRs in order to keep the commit history clean.

### CODE REVIEW

**Comment by falaki** [comment]
Thanks. I will this PR to merge itself.

File: dev/merge_pr.py [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Utility for creating well-formed pull request merges and pushing them to Apache.
+#   usage: ./apache-pr-merge.py    (see config env vars below)
```
Comment (falaki) [inline dev/merge_pr.py]: usage: ./dev/merge-pr.py

File: dev/merge_pr.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+#
+# This utility assumes you already have local a Spark git folder and that you
```
Comment (falaki) [inline dev/merge_pr.py]: Please remove these references to spark.

File: dev/merge_pr.py [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+import sys
+import urllib2
+
+try:
+    import jira.client
+    JIRA_IMPORTED = True
+except ImportError:
+    JIRA_IMPORTED = False
+
+# Location of your Spark git development area
```
Comment (falaki) [inline dev/merge_pr.py]: Are these needed any more?

File: dev/merge_pr.py [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+def get_json(url):
+    try:
+        request = urllib2.Request(url)
+        if GITHUB_OAUTH_KEY:
+            request.add_header('Authorization', 'token %s' % GITHUB_OAUTH_KEY)
+        return json.load(urllib2.urlopen(request))
+    except urllib2.HTTPError as e:
+        if "X-RateLimit-Remaining" in e.headers and e.headers["X-RateLimit-Remaining"] == '0':
+            print "Exceeded the GitHub API rate limit; see the instructions in " + \
+                  "dev/merge_spark_pr.py to configure an OAuth token for making authenticated " + \
```
Comment (falaki) [inline dev/merge_pr.py]: dev/merge-pr.py

File: dev/merge_pr.py (hunk 2 above)
Comment (JoshRosen) [inline dev/merge_pr.py]: FWIW this is a quick-and-dirty port of the merge script, only intended to be used by Databricks folks and not anyone else. Hence there's a lot of possible cleanup that we chose not to do when adding it to the Avro and Redshift libraries.

File: dev/merge_pr.py [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+    latest_branch = sorted(branch_names, reverse=True)[0]
+
+    pr_num = raw_input("Which pull request would you like to merge? (e.g. 34): ")
+    pr = get_json("%s/pulls/%s" % (GITHUB_API_BASE, pr_num))
+    pr_events = get_json("%s/issues/%s/events" % (GITHUB_API_BASE, pr_num))
+
+    url = pr["url"]
+
+    # Decide whether to use the modified title or not
+    modified_title = standardize_jira_ref(pr["title"])
```
Comment (falaki) [inline dev/merge_pr.py]: Is it safe to make this call with spark-csv?

**Comment by JoshRosen** [comment]
Honestly, I'd rather just merge this now and defer the cleanup to later since we're going to be the only ones who use this script.

File: dev/merge_pr.py (hunk 5 above)
Comment (JoshRosen) [inline dev/merge_pr.py]: Just hit "no" when asked to update JIRAs.

Comment (JoshRosen) [inline dev/merge_pr.py]: And yeah, this is safe since we've been doing it with the Redshift and Avro libraries.

**Pull request closed without merging**